// @azure/arm-appinsights/dist/react-native/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link ApplicationType} that the service accepts. */
export var KnownApplicationType;
(function (KnownApplicationType) {
    /** Web */
    KnownApplicationType["Web"] = "web";
    /** Other */
    KnownApplicationType["Other"] = "other";
})(KnownApplicationType || (KnownApplicationType = {}));
/** Known values of {@link FlowType} that the service accepts. */
export var KnownFlowType;
(function (KnownFlowType) {
    /** Bluefield */
    KnownFlowType["Bluefield"] = "Bluefield";
})(KnownFlowType || (KnownFlowType = {}));
/** Known values of {@link RequestSource} that the service accepts. */
export var KnownRequestSource;
(function (KnownRequestSource) {
    /** Rest */
    KnownRequestSource["Rest"] = "rest";
})(KnownRequestSource || (KnownRequestSource = {}));
/** Known values of {@link PublicNetworkAccessType} that the service accepts. */
export var KnownPublicNetworkAccessType;
(function (KnownPublicNetworkAccessType) {
    /** Enables connectivity to Application Insights through public DNS. */
    KnownPublicNetworkAccessType["Enabled"] = "Enabled";
    /** Disables public connectivity to Application Insights through public DNS. */
    KnownPublicNetworkAccessType["Disabled"] = "Disabled";
})(KnownPublicNetworkAccessType || (KnownPublicNetworkAccessType = {}));
/** Known values of {@link IngestionMode} that the service accepts. */
export var KnownIngestionMode;
(function (KnownIngestionMode) {
    /** ApplicationInsights */
    KnownIngestionMode["ApplicationInsights"] = "ApplicationInsights";
    /** ApplicationInsightsWithDiagnosticSettings */
    KnownIngestionMode["ApplicationInsightsWithDiagnosticSettings"] = "ApplicationInsightsWithDiagnosticSettings";
    /** LogAnalytics */
    KnownIngestionMode["LogAnalytics"] = "LogAnalytics";
})(KnownIngestionMode || (KnownIngestionMode = {}));
/** Known values of {@link PurgeState} that the service accepts. */
export var KnownPurgeState;
(function (KnownPurgeState) {
    /** Pending */
    KnownPurgeState["Pending"] = "pending";
    /** Completed */
    KnownPurgeState["Completed"] = "completed";
})(KnownPurgeState || (KnownPurgeState = {}));
/** Known values of {@link FavoriteSourceType} that the service accepts. */
export var KnownFavoriteSourceType;
(function (KnownFavoriteSourceType) {
    /** Retention */
    KnownFavoriteSourceType["Retention"] = "retention";
    /** Notebook */
    KnownFavoriteSourceType["Notebook"] = "notebook";
    /** Sessions */
    KnownFavoriteSourceType["Sessions"] = "sessions";
    /** Events */
    KnownFavoriteSourceType["Events"] = "events";
    /** Userflows */
    KnownFavoriteSourceType["Userflows"] = "userflows";
    /** Funnel */
    KnownFavoriteSourceType["Funnel"] = "funnel";
    /** Impact */
    KnownFavoriteSourceType["Impact"] = "impact";
    /** Segmentation */
    KnownFavoriteSourceType["Segmentation"] = "segmentation";
})(KnownFavoriteSourceType || (KnownFavoriteSourceType = {}));
/** Known values of {@link ItemScopePath} that the service accepts. */
export var KnownItemScopePath;
(function (KnownItemScopePath) {
    /** AnalyticsItems */
    KnownItemScopePath["AnalyticsItems"] = "analyticsItems";
    /** MyanalyticsItems */
    KnownItemScopePath["MyanalyticsItems"] = "myanalyticsItems";
})(KnownItemScopePath || (KnownItemScopePath = {}));
/** Known values of {@link ItemScope} that the service accepts. */
export var KnownItemScope;
(function (KnownItemScope) {
    /** Shared */
    KnownItemScope["Shared"] = "shared";
    /** User */
    KnownItemScope["User"] = "user";
})(KnownItemScope || (KnownItemScope = {}));
/** Known values of {@link ItemTypeParameter} that the service accepts. */
export var KnownItemTypeParameter;
(function (KnownItemTypeParameter) {
    /** None */
    KnownItemTypeParameter["None"] = "none";
    /** Query */
    KnownItemTypeParameter["Query"] = "query";
    /** Function */
    KnownItemTypeParameter["Function"] = "function";
    /** Folder */
    KnownItemTypeParameter["Folder"] = "folder";
    /** Recent */
    KnownItemTypeParameter["Recent"] = "recent";
})(KnownItemTypeParameter || (KnownItemTypeParameter = {}));
/** Known values of {@link ItemType} that the service accepts. */
export var KnownItemType;
(function (KnownItemType) {
    /** None */
    KnownItemType["None"] = "none";
    /** Query */
    KnownItemType["Query"] = "query";
    /** Recent */
    KnownItemType["Recent"] = "recent";
    /** Function */
    KnownItemType["Function"] = "function";
})(KnownItemType || (KnownItemType = {}));
/** Known values of {@link CategoryType} that the service accepts. */
export var KnownCategoryType;
(function (KnownCategoryType) {
    /** Workbook */
    KnownCategoryType["Workbook"] = "workbook";
    /** TSG */
    KnownCategoryType["TSG"] = "TSG";
    /** Performance */
    KnownCategoryType["Performance"] = "performance";
    /** Retention */
    KnownCategoryType["Retention"] = "retention";
})(KnownCategoryType || (KnownCategoryType = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link ManagedServiceIdentityType} that the service accepts. */
export var KnownManagedServiceIdentityType;
(function (KnownManagedServiceIdentityType) {
    /** None */
    KnownManagedServiceIdentityType["None"] = "None";
    /** SystemAssigned */
    KnownManagedServiceIdentityType["SystemAssigned"] = "SystemAssigned";
    /** UserAssigned */
    KnownManagedServiceIdentityType["UserAssigned"] = "UserAssigned";
    /** SystemAssignedUserAssigned */
    KnownManagedServiceIdentityType["SystemAssignedUserAssigned"] = "SystemAssigned,UserAssigned";
})(KnownManagedServiceIdentityType || (KnownManagedServiceIdentityType = {}));
/** Known values of {@link WorkbookSharedTypeKind} that the service accepts. */
export var KnownWorkbookSharedTypeKind;
(function (KnownWorkbookSharedTypeKind) {
    /** Shared */
    KnownWorkbookSharedTypeKind["Shared"] = "shared";
})(KnownWorkbookSharedTypeKind || (KnownWorkbookSharedTypeKind = {}));
/** Known values of {@link WorkbookUpdateSharedTypeKind} that the service accepts. */
export var KnownWorkbookUpdateSharedTypeKind;
(function (KnownWorkbookUpdateSharedTypeKind) {
    /** Shared */
    KnownWorkbookUpdateSharedTypeKind["Shared"] = "shared";
})(KnownWorkbookUpdateSharedTypeKind || (KnownWorkbookUpdateSharedTypeKind = {}));
/** Known values of {@link StorageType} that the service accepts. */
export var KnownStorageType;
(function (KnownStorageType) {
    /** ServiceProfiler */
    KnownStorageType["ServiceProfiler"] = "ServiceProfiler";
})(KnownStorageType || (KnownStorageType = {}));
