// @azure/arm-costmanagement/dist/commonjs/operations/generateDetailedCostReport.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenerateDetailedCostReportImpl = void 0;
const tslib_1 = require("tslib");
const coreClient = tslib_1.__importStar(require("@azure/core-client"));
const Mappers = tslib_1.__importStar(require("../models/mappers.js"));
const Parameters = tslib_1.__importStar(require("../models/parameters.js"));
const core_lro_1 = require("@azure/core-lro");
const lroImpl_js_1 = require("../lroImpl.js");
/** Class containing GenerateDetailedCostReport operations. */
class GenerateDetailedCostReportImpl {
    client;
    /**
     * Initialize a new instance of the class GenerateDetailedCostReport class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Generates the detailed cost report for provided date range, billing period(only enterprise
     * customers) or Invoice ID asynchronously at a certain scope. Call returns a 202 with header
     * Azure-Consumption-AsyncOperation providing a link to the operation created. A call on the operation
     * will provide the status and if the operation is completed the blob file where generated detailed
     * cost report is being stored.
     * @param scope The ARM Resource ID for subscription, resource group, billing account, or other billing
     *              scopes. For details, see https://aka.ms/costmgmt/scopes.
     * @param parameters Parameters supplied to the Create detailed cost report operation.
     * @param options The options parameters.
     */
    async beginCreateOperation(scope, parameters, options) {
        const directSendOperation = async (args, spec) => {
            return this.client.sendOperationRequest(args, spec);
        };
        const sendOperationFn = async (args, spec) => {
            let currentRawResponse = undefined;
            const providedCallback = args.options?.onResponse;
            const callback = (rawResponse, flatResponse) => {
                currentRawResponse = rawResponse;
                providedCallback?.(rawResponse, flatResponse);
            };
            const updatedArgs = {
                ...args,
                options: {
                    ...args.options,
                    onResponse: callback
                }
            };
            const flatResponse = await directSendOperation(updatedArgs, spec);
            return {
                flatResponse,
                rawResponse: {
                    statusCode: currentRawResponse.status,
                    body: currentRawResponse.parsedBody,
                    headers: currentRawResponse.headers.toJSON()
                }
            };
        };
        const lro = (0, lroImpl_js_1.createLroSpec)({
            sendOperationFn,
            args: { scope, parameters, options },
            spec: createOperationOperationSpec
        });
        const poller = await (0, core_lro_1.createHttpPoller)(lro, {
            restoreFrom: options?.resumeFrom,
            intervalInMs: options?.updateIntervalInMs,
            resourceLocationConfig: "location"
        });
        await poller.poll();
        return poller;
    }
    /**
     * Generates the detailed cost report for provided date range, billing period(only enterprise
     * customers) or Invoice ID asynchronously at a certain scope. Call returns a 202 with header
     * Azure-Consumption-AsyncOperation providing a link to the operation created. A call on the operation
     * will provide the status and if the operation is completed the blob file where generated detailed
     * cost report is being stored.
     * @param scope The ARM Resource ID for subscription, resource group, billing account, or other billing
     *              scopes. For details, see https://aka.ms/costmgmt/scopes.
     * @param parameters Parameters supplied to the Create detailed cost report operation.
     * @param options The options parameters.
     */
    async beginCreateOperationAndWait(scope, parameters, options) {
        const poller = await this.beginCreateOperation(scope, parameters, options);
        return poller.pollUntilDone();
    }
}
exports.GenerateDetailedCostReportImpl = GenerateDetailedCostReportImpl;
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const createOperationOperationSpec = {
    path: "/{scope}/providers/Microsoft.CostManagement/generateDetailedCostReport",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.GenerateDetailedCostReportOperationResult
        },
        201: {
            bodyMapper: Mappers.GenerateDetailedCostReportOperationResult
        },
        202: {
            bodyMapper: Mappers.GenerateDetailedCostReportOperationResult
        },
        204: {
            bodyMapper: Mappers.GenerateDetailedCostReportOperationResult
        },
        default: {
            bodyMapper: Mappers.GenerateDetailedCostReportErrorResponse
        }
    },
    requestBody: Parameters.parameters6,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [Parameters.$host, Parameters.scope1],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
