// @azure/arm-appinsights/dist/commonjs/operations/workItemConfigurations.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkItemConfigurationsImpl = void 0;
const tslib_1 = require("tslib");
const coreClient = tslib_1.__importStar(require("@azure/core-client"));
const Mappers = tslib_1.__importStar(require("../models/mappers.js"));
const Parameters = tslib_1.__importStar(require("../models/parameters.js"));
/// <reference lib="esnext.asynciterable" />
/** Class containing WorkItemConfigurations operations. */
class WorkItemConfigurationsImpl {
    /**
     * Initialize a new instance of the class WorkItemConfigurations class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Gets the list work item configurations that exist for the application
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param resourceName The name of the Application Insights component resource.
     * @param options The options parameters.
     */
    list(resourceGroupName, resourceName, options) {
        const iter = this.listPagingAll(resourceGroupName, resourceName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings === null || settings === void 0 ? void 0 : settings.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(resourceGroupName, resourceName, options, settings);
            },
        };
    }
    listPagingPage(resourceGroupName, resourceName, options, _settings) {
        return tslib_1.__asyncGenerator(this, arguments, function* listPagingPage_1() {
            let result;
            result = yield tslib_1.__await(this._list(resourceGroupName, resourceName, options));
            yield yield tslib_1.__await(result.value || []);
        });
    }
    listPagingAll(resourceGroupName, resourceName, options) {
        return tslib_1.__asyncGenerator(this, arguments, function* listPagingAll_1() {
            var _a, e_1, _b, _c;
            try {
                for (var _d = true, _e = tslib_1.__asyncValues(this.listPagingPage(resourceGroupName, resourceName, options)), _f; _f = yield tslib_1.__await(_e.next()), _a = _f.done, !_a; _d = true) {
                    _c = _f.value;
                    _d = false;
                    const page = _c;
                    yield tslib_1.__await(yield* tslib_1.__asyncDelegator(tslib_1.__asyncValues(page)));
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (!_d && !_a && (_b = _e.return)) yield tslib_1.__await(_b.call(_e));
                }
                finally { if (e_1) throw e_1.error; }
            }
        });
    }
    /**
     * Gets the list work item configurations that exist for the application
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param resourceName The name of the Application Insights component resource.
     * @param options The options parameters.
     */
    _list(resourceGroupName, resourceName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, resourceName, options }, listOperationSpec);
    }
    /**
     * Create a work item configuration for an Application Insights component.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param resourceName The name of the Application Insights component resource.
     * @param workItemConfigurationProperties Properties that need to be specified to create a work item
     *                                        configuration of a Application Insights component.
     * @param options The options parameters.
     */
    create(resourceGroupName, resourceName, workItemConfigurationProperties, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            resourceName,
            workItemConfigurationProperties,
            options,
        }, createOperationSpec);
    }
    /**
     * Gets default work item configurations that exist for the application
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param resourceName The name of the Application Insights component resource.
     * @param options The options parameters.
     */
    getDefault(resourceGroupName, resourceName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, resourceName, options }, getDefaultOperationSpec);
    }
    /**
     * Delete a work item configuration of an Application Insights component.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param resourceName The name of the Application Insights component resource.
     * @param workItemConfigId The unique work item configuration Id. This can be either friendly name of
     *                         connector as defined in connector configuration
     * @param options The options parameters.
     */
    delete(resourceGroupName, resourceName, workItemConfigId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, resourceName, workItemConfigId, options }, deleteOperationSpec);
    }
    /**
     * Gets specified work item configuration for an Application Insights component.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param resourceName The name of the Application Insights component resource.
     * @param workItemConfigId The unique work item configuration Id. This can be either friendly name of
     *                         connector as defined in connector configuration
     * @param options The options parameters.
     */
    getItem(resourceGroupName, resourceName, workItemConfigId, options) {
        return this.client.sendOperationRequest({ resourceGroupName, resourceName, workItemConfigId, options }, getItemOperationSpec);
    }
    /**
     * Update a work item configuration for an Application Insights component.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param resourceName The name of the Application Insights component resource.
     * @param workItemConfigId The unique work item configuration Id. This can be either friendly name of
     *                         connector as defined in connector configuration
     * @param workItemConfigurationProperties Properties that need to be specified to update a work item
     *                                        configuration for this Application Insights component.
     * @param options The options parameters.
     */
    updateItem(resourceGroupName, resourceName, workItemConfigId, workItemConfigurationProperties, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            resourceName,
            workItemConfigId,
            workItemConfigurationProperties,
            options,
        }, updateItemOperationSpec);
    }
}
exports.WorkItemConfigurationsImpl = WorkItemConfigurationsImpl;
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/components/{resourceName}/WorkItemConfigs",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.WorkItemConfigurationsListResult,
        },
        default: {
            bodyMapper: Mappers.WorkItemConfigurationError,
        },
    },
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const createOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/components/{resourceName}/WorkItemConfigs",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.WorkItemConfiguration,
        },
    },
    requestBody: Parameters.workItemConfigurationProperties,
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName,
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer,
};
const getDefaultOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/components/{resourceName}/DefaultWorkItemConfig",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.WorkItemConfiguration,
        },
    },
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/components/{resourceName}/WorkItemConfigs/{workItemConfigId}",
    httpMethod: "DELETE",
    responses: { 200: {} },
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName,
        Parameters.workItemConfigId,
    ],
    serializer,
};
const getItemOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/components/{resourceName}/WorkItemConfigs/{workItemConfigId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.WorkItemConfiguration,
        },
    },
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName,
        Parameters.workItemConfigId,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const updateItemOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/components/{resourceName}/WorkItemConfigs/{workItemConfigId}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.WorkItemConfiguration,
        },
    },
    requestBody: Parameters.workItemConfigurationProperties,
    queryParameters: [Parameters.apiVersion1],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.resourceName,
        Parameters.workItemConfigId,
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer,
};
