// @azure/arm-appinsights/dist/commonjs/models/mappers.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorFieldContract = exports.ErrorResponse = exports.OperationDisplay = exports.Operation = exports.OperationListResult = exports.ApplicationInsightsComponentAnalyticsItemProperties = exports.ApplicationInsightsComponentAnalyticsItem = exports.WebtestsResource = exports.WebTestPropertiesValidationRulesContentValidation = exports.WebTestPropertiesValidationRules = exports.HeaderField = exports.WebTestPropertiesRequest = exports.WebTestPropertiesConfiguration = exports.WebTestGeolocation = exports.WebTestListResult = exports.ApplicationInsightsComponentWebTestLocation = exports.ApplicationInsightsWebTestLocationsListResult = exports.ApplicationInsightsComponentFavorite = exports.WorkItemCreateConfiguration = exports.WorkItemConfigurationError = exports.WorkItemConfiguration = exports.WorkItemConfigurationsListResult = exports.ApplicationInsightsComponentProactiveDetectionConfigurationRuleDefinitions = exports.ApplicationInsightsComponentProactiveDetectionConfiguration = exports.ApplicationInsightsComponentFeatureCapability = exports.ApplicationInsightsComponentFeature = exports.ApplicationInsightsComponentAvailableFeatures = exports.ApplicationInsightsComponentFeatureCapabilities = exports.ApplicationInsightsComponentQuotaStatus = exports.ApplicationInsightsComponentDataVolumeCap = exports.ApplicationInsightsComponentBillingFeatures = exports.ApplicationInsightsComponentExportRequest = exports.ApplicationInsightsComponentExportConfiguration = exports.APIKeyRequest = exports.ApplicationInsightsComponentAPIKey = exports.ApplicationInsightsComponentAPIKeyListResult = exports.InnerError = exports.AnnotationError = exports.Annotation = exports.AnnotationsListResult = exports.ComponentPurgeStatusResponse = exports.ComponentPurgeResponse = exports.ComponentPurgeBodyFilters = exports.ComponentPurgeBody = exports.TagsResource = exports.ErrorResponseComponentsError = exports.ErrorResponseComponents = exports.ComponentsResource = exports.PrivateLinkScopedResource = exports.ApplicationInsightsComponentListResult = void 0;
exports.DeletedWorkbook = exports.Workbook = exports.DeletedWorkbookResource = exports.ComponentLinkedStorageAccounts = exports.WorkbookResource = exports.TrackedResourceAutoGenerated = exports.ProxyResource = exports.TrackedResource = exports.WorkbookResourceIdentity = exports.WorkbookTemplate = exports.WebTest = exports.ApplicationInsightsComponent = exports.OperationInfo = exports.OperationLive = exports.OperationsListResult = exports.DeletedWorkbookInnerErrorTrace = exports.DeletedWorkbookErrorDefinition = exports.DeletedWorkbookError = exports.ResourceAutoGenerated = exports.DeletedWorkbooksListResult = exports.ComponentLinkedStorageAccountsPatch = exports.ErrorResponseLinkedStorageError = exports.ErrorResponseLinkedStorage = exports.LiveTokenResponse = exports.WorkbookUpdateParameters = exports.WorkbookInnerErrorTrace = exports.WorkbookErrorDefinition = exports.WorkbookError = exports.Resource = exports.UserAssignedIdentity = exports.ManagedServiceIdentity = exports.SystemData = exports.WorkbooksListResult = exports.WorkbookTemplateUpdateParameters = exports.WorkbookTemplateErrorFieldContract = exports.WorkbookTemplateErrorBody = exports.WorkbookTemplateError = exports.WorkbookTemplateResource = exports.WorkbookTemplateLocalizedGallery = exports.WorkbookTemplateGallery = exports.WorkbookTemplatesListResult = void 0;
exports.ApplicationInsightsComponentListResult = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ApplicationInsightsComponent",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.PrivateLinkScopedResource = {
    type: {
        name: "Composite",
        className: "PrivateLinkScopedResource",
        modelProperties: {
            resourceId: {
                serializedName: "ResourceId",
                type: {
                    name: "String",
                },
            },
            scopeId: {
                serializedName: "ScopeId",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ComponentsResource = {
    type: {
        name: "Composite",
        className: "ComponentsResource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String",
                },
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
exports.ErrorResponseComponents = {
    type: {
        name: "Composite",
        className: "ErrorResponseComponents",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorResponseComponentsError",
                },
            },
        },
    },
};
exports.ErrorResponseComponentsError = {
    type: {
        name: "Composite",
        className: "ErrorResponseComponentsError",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.TagsResource = {
    type: {
        name: "Composite",
        className: "TagsResource",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
exports.ComponentPurgeBody = {
    type: {
        name: "Composite",
        className: "ComponentPurgeBody",
        modelProperties: {
            table: {
                serializedName: "table",
                required: true,
                type: {
                    name: "String",
                },
            },
            filters: {
                serializedName: "filters",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ComponentPurgeBodyFilters",
                        },
                    },
                },
            },
        },
    },
};
exports.ComponentPurgeBodyFilters = {
    type: {
        name: "Composite",
        className: "ComponentPurgeBodyFilters",
        modelProperties: {
            column: {
                serializedName: "column",
                type: {
                    name: "String",
                },
            },
            operator: {
                serializedName: "operator",
                type: {
                    name: "String",
                },
            },
            value: {
                serializedName: "value",
                type: {
                    name: "any",
                },
            },
            key: {
                serializedName: "key",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ComponentPurgeResponse = {
    type: {
        name: "Composite",
        className: "ComponentPurgeResponse",
        modelProperties: {
            operationId: {
                serializedName: "operationId",
                required: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ComponentPurgeStatusResponse = {
    type: {
        name: "Composite",
        className: "ComponentPurgeStatusResponse",
        modelProperties: {
            status: {
                serializedName: "status",
                required: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.AnnotationsListResult = {
    type: {
        name: "Composite",
        className: "AnnotationsListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Annotation",
                        },
                    },
                },
            },
        },
    },
};
exports.Annotation = {
    type: {
        name: "Composite",
        className: "Annotation",
        modelProperties: {
            annotationName: {
                serializedName: "AnnotationName",
                type: {
                    name: "String",
                },
            },
            category: {
                serializedName: "Category",
                type: {
                    name: "String",
                },
            },
            eventTime: {
                serializedName: "EventTime",
                type: {
                    name: "DateTime",
                },
            },
            id: {
                serializedName: "Id",
                type: {
                    name: "String",
                },
            },
            properties: {
                serializedName: "Properties",
                type: {
                    name: "String",
                },
            },
            relatedAnnotation: {
                defaultValue: "null",
                serializedName: "RelatedAnnotation",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.AnnotationError = {
    type: {
        name: "Composite",
        className: "AnnotationError",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String",
                },
            },
            innererror: {
                serializedName: "innererror",
                type: {
                    name: "Composite",
                    className: "InnerError",
                },
            },
        },
    },
};
exports.InnerError = {
    type: {
        name: "Composite",
        className: "InnerError",
        modelProperties: {
            diagnosticcontext: {
                serializedName: "diagnosticcontext",
                type: {
                    name: "String",
                },
            },
            time: {
                serializedName: "time",
                type: {
                    name: "DateTime",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentAPIKeyListResult = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentAPIKeyListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ApplicationInsightsComponentAPIKey",
                        },
                    },
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentAPIKey = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentAPIKey",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            apiKey: {
                serializedName: "apiKey",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            createdDate: {
                serializedName: "createdDate",
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            linkedReadProperties: {
                serializedName: "linkedReadProperties",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            linkedWriteProperties: {
                serializedName: "linkedWriteProperties",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
exports.APIKeyRequest = {
    type: {
        name: "Composite",
        className: "APIKeyRequest",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            linkedReadProperties: {
                serializedName: "linkedReadProperties",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            linkedWriteProperties: {
                serializedName: "linkedWriteProperties",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentExportConfiguration = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentExportConfiguration",
        modelProperties: {
            exportId: {
                serializedName: "ExportId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            instrumentationKey: {
                serializedName: "InstrumentationKey",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            recordTypes: {
                serializedName: "RecordTypes",
                type: {
                    name: "String",
                },
            },
            applicationName: {
                serializedName: "ApplicationName",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            subscriptionId: {
                serializedName: "SubscriptionId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            resourceGroup: {
                serializedName: "ResourceGroup",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            destinationStorageSubscriptionId: {
                serializedName: "DestinationStorageSubscriptionId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            destinationStorageLocationId: {
                serializedName: "DestinationStorageLocationId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            destinationAccountId: {
                serializedName: "DestinationAccountId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            destinationType: {
                serializedName: "DestinationType",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            isUserEnabled: {
                serializedName: "IsUserEnabled",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            lastUserUpdate: {
                serializedName: "LastUserUpdate",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            notificationQueueEnabled: {
                serializedName: "NotificationQueueEnabled",
                type: {
                    name: "String",
                },
            },
            exportStatus: {
                serializedName: "ExportStatus",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            lastSuccessTime: {
                serializedName: "LastSuccessTime",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            lastGapTime: {
                serializedName: "LastGapTime",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            permanentErrorReason: {
                serializedName: "PermanentErrorReason",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            storageName: {
                serializedName: "StorageName",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            containerName: {
                serializedName: "ContainerName",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentExportRequest = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentExportRequest",
        modelProperties: {
            recordTypes: {
                serializedName: "RecordTypes",
                type: {
                    name: "String",
                },
            },
            destinationType: {
                serializedName: "DestinationType",
                type: {
                    name: "String",
                },
            },
            destinationAddress: {
                serializedName: "DestinationAddress",
                type: {
                    name: "String",
                },
            },
            isEnabled: {
                serializedName: "IsEnabled",
                type: {
                    name: "String",
                },
            },
            notificationQueueEnabled: {
                serializedName: "NotificationQueueEnabled",
                type: {
                    name: "String",
                },
            },
            notificationQueueUri: {
                serializedName: "NotificationQueueUri",
                type: {
                    name: "String",
                },
            },
            destinationStorageSubscriptionId: {
                serializedName: "DestinationStorageSubscriptionId",
                type: {
                    name: "String",
                },
            },
            destinationStorageLocationId: {
                serializedName: "DestinationStorageLocationId",
                type: {
                    name: "String",
                },
            },
            destinationAccountId: {
                serializedName: "DestinationAccountId",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentBillingFeatures = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentBillingFeatures",
        modelProperties: {
            dataVolumeCap: {
                serializedName: "DataVolumeCap",
                type: {
                    name: "Composite",
                    className: "ApplicationInsightsComponentDataVolumeCap",
                },
            },
            currentBillingFeatures: {
                serializedName: "CurrentBillingFeatures",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentDataVolumeCap = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentDataVolumeCap",
        modelProperties: {
            cap: {
                serializedName: "Cap",
                type: {
                    name: "Number",
                },
            },
            resetTime: {
                serializedName: "ResetTime",
                readOnly: true,
                type: {
                    name: "Number",
                },
            },
            warningThreshold: {
                serializedName: "WarningThreshold",
                type: {
                    name: "Number",
                },
            },
            stopSendNotificationWhenHitThreshold: {
                serializedName: "StopSendNotificationWhenHitThreshold",
                type: {
                    name: "Boolean",
                },
            },
            stopSendNotificationWhenHitCap: {
                serializedName: "StopSendNotificationWhenHitCap",
                type: {
                    name: "Boolean",
                },
            },
            maxHistoryCap: {
                serializedName: "MaxHistoryCap",
                readOnly: true,
                type: {
                    name: "Number",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentQuotaStatus = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentQuotaStatus",
        modelProperties: {
            appId: {
                serializedName: "AppId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            shouldBeThrottled: {
                serializedName: "ShouldBeThrottled",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            expirationTime: {
                serializedName: "ExpirationTime",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentFeatureCapabilities = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentFeatureCapabilities",
        modelProperties: {
            supportExportData: {
                serializedName: "SupportExportData",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            burstThrottlePolicy: {
                serializedName: "BurstThrottlePolicy",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            metadataClass: {
                serializedName: "MetadataClass",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            liveStreamMetrics: {
                serializedName: "LiveStreamMetrics",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            applicationMap: {
                serializedName: "ApplicationMap",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            workItemIntegration: {
                serializedName: "WorkItemIntegration",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            powerBIIntegration: {
                serializedName: "PowerBIIntegration",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            openSchema: {
                serializedName: "OpenSchema",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            proactiveDetection: {
                serializedName: "ProactiveDetection",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            analyticsIntegration: {
                serializedName: "AnalyticsIntegration",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            multipleStepWebTest: {
                serializedName: "MultipleStepWebTest",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            apiAccessLevel: {
                serializedName: "ApiAccessLevel",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            trackingType: {
                serializedName: "TrackingType",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            dailyCap: {
                serializedName: "DailyCap",
                readOnly: true,
                type: {
                    name: "Number",
                },
            },
            dailyCapResetTime: {
                serializedName: "DailyCapResetTime",
                readOnly: true,
                type: {
                    name: "Number",
                },
            },
            throttleRate: {
                serializedName: "ThrottleRate",
                readOnly: true,
                type: {
                    name: "Number",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentAvailableFeatures = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentAvailableFeatures",
        modelProperties: {
            result: {
                serializedName: "Result",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ApplicationInsightsComponentFeature",
                        },
                    },
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentFeature = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentFeature",
        modelProperties: {
            featureName: {
                serializedName: "FeatureName",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            meterId: {
                serializedName: "MeterId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            meterRateFrequency: {
                serializedName: "MeterRateFrequency",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            resouceId: {
                serializedName: "ResouceId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            isHidden: {
                serializedName: "IsHidden",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            capabilities: {
                serializedName: "Capabilities",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ApplicationInsightsComponentFeatureCapability",
                        },
                    },
                },
            },
            title: {
                serializedName: "Title",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            isMainFeature: {
                serializedName: "IsMainFeature",
                readOnly: true,
                type: {
                    name: "Boolean",
                },
            },
            supportedAddonFeatures: {
                serializedName: "SupportedAddonFeatures",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentFeatureCapability = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentFeatureCapability",
        modelProperties: {
            name: {
                serializedName: "Name",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            description: {
                serializedName: "Description",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            value: {
                serializedName: "Value",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            unit: {
                serializedName: "Unit",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            meterId: {
                serializedName: "MeterId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            meterRateFrequency: {
                serializedName: "MeterRateFrequency",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentProactiveDetectionConfiguration = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentProactiveDetectionConfiguration",
        modelProperties: {
            name: {
                serializedName: "Name",
                type: {
                    name: "String",
                },
            },
            enabled: {
                serializedName: "Enabled",
                type: {
                    name: "Boolean",
                },
            },
            sendEmailsToSubscriptionOwners: {
                serializedName: "SendEmailsToSubscriptionOwners",
                type: {
                    name: "Boolean",
                },
            },
            customEmails: {
                serializedName: "CustomEmails",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            lastUpdatedTime: {
                serializedName: "LastUpdatedTime",
                type: {
                    name: "String",
                },
            },
            ruleDefinitions: {
                serializedName: "RuleDefinitions",
                type: {
                    name: "Composite",
                    className: "ApplicationInsightsComponentProactiveDetectionConfigurationRuleDefinitions",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentProactiveDetectionConfigurationRuleDefinitions = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentProactiveDetectionConfigurationRuleDefinitions",
        modelProperties: {
            name: {
                serializedName: "Name",
                type: {
                    name: "String",
                },
            },
            displayName: {
                serializedName: "DisplayName",
                type: {
                    name: "String",
                },
            },
            description: {
                serializedName: "Description",
                type: {
                    name: "String",
                },
            },
            helpUrl: {
                serializedName: "HelpUrl",
                type: {
                    name: "String",
                },
            },
            isHidden: {
                serializedName: "IsHidden",
                type: {
                    name: "Boolean",
                },
            },
            isEnabledByDefault: {
                serializedName: "IsEnabledByDefault",
                type: {
                    name: "Boolean",
                },
            },
            isInPreview: {
                serializedName: "IsInPreview",
                type: {
                    name: "Boolean",
                },
            },
            supportsEmailNotifications: {
                serializedName: "SupportsEmailNotifications",
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
exports.WorkItemConfigurationsListResult = {
    type: {
        name: "Composite",
        className: "WorkItemConfigurationsListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WorkItemConfiguration",
                        },
                    },
                },
            },
        },
    },
};
exports.WorkItemConfiguration = {
    type: {
        name: "Composite",
        className: "WorkItemConfiguration",
        modelProperties: {
            connectorId: {
                serializedName: "ConnectorId",
                type: {
                    name: "String",
                },
            },
            configDisplayName: {
                serializedName: "ConfigDisplayName",
                type: {
                    name: "String",
                },
            },
            isDefault: {
                serializedName: "IsDefault",
                type: {
                    name: "Boolean",
                },
            },
            id: {
                serializedName: "Id",
                type: {
                    name: "String",
                },
            },
            configProperties: {
                serializedName: "ConfigProperties",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WorkItemConfigurationError = {
    type: {
        name: "Composite",
        className: "WorkItemConfigurationError",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String",
                },
            },
            innererror: {
                serializedName: "innererror",
                type: {
                    name: "Composite",
                    className: "InnerError",
                },
            },
        },
    },
};
exports.WorkItemCreateConfiguration = {
    type: {
        name: "Composite",
        className: "WorkItemCreateConfiguration",
        modelProperties: {
            connectorId: {
                serializedName: "ConnectorId",
                type: {
                    name: "String",
                },
            },
            connectorDataConfiguration: {
                serializedName: "ConnectorDataConfiguration",
                type: {
                    name: "String",
                },
            },
            validateOnly: {
                serializedName: "ValidateOnly",
                type: {
                    name: "Boolean",
                },
            },
            workItemProperties: {
                serializedName: "WorkItemProperties",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentFavorite = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentFavorite",
        modelProperties: {
            name: {
                serializedName: "Name",
                type: {
                    name: "String",
                },
            },
            config: {
                serializedName: "Config",
                type: {
                    name: "String",
                },
            },
            version: {
                serializedName: "Version",
                type: {
                    name: "String",
                },
            },
            favoriteId: {
                serializedName: "FavoriteId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            favoriteType: {
                serializedName: "FavoriteType",
                type: {
                    name: "Enum",
                    allowedValues: ["shared", "user"],
                },
            },
            sourceType: {
                serializedName: "SourceType",
                type: {
                    name: "String",
                },
            },
            timeModified: {
                serializedName: "TimeModified",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            tags: {
                serializedName: "Tags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            category: {
                serializedName: "Category",
                type: {
                    name: "String",
                },
            },
            isGeneratedFromTemplate: {
                serializedName: "IsGeneratedFromTemplate",
                type: {
                    name: "Boolean",
                },
            },
            userId: {
                serializedName: "UserId",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ApplicationInsightsWebTestLocationsListResult = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsWebTestLocationsListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ApplicationInsightsComponentWebTestLocation",
                        },
                    },
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentWebTestLocation = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentWebTestLocation",
        modelProperties: {
            displayName: {
                serializedName: "DisplayName",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            tag: {
                serializedName: "Tag",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebTestListResult = {
    type: {
        name: "Composite",
        className: "WebTestListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                required: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WebTest",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebTestGeolocation = {
    type: {
        name: "Composite",
        className: "WebTestGeolocation",
        modelProperties: {
            location: {
                serializedName: "Id",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebTestPropertiesConfiguration = {
    type: {
        name: "Composite",
        className: "WebTestPropertiesConfiguration",
        modelProperties: {
            webTest: {
                serializedName: "WebTest",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebTestPropertiesRequest = {
    type: {
        name: "Composite",
        className: "WebTestPropertiesRequest",
        modelProperties: {
            requestUrl: {
                serializedName: "RequestUrl",
                type: {
                    name: "String",
                },
            },
            headers: {
                serializedName: "Headers",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "HeaderField",
                        },
                    },
                },
            },
            httpVerb: {
                serializedName: "HttpVerb",
                type: {
                    name: "String",
                },
            },
            requestBody: {
                serializedName: "RequestBody",
                type: {
                    name: "String",
                },
            },
            parseDependentRequests: {
                serializedName: "ParseDependentRequests",
                type: {
                    name: "Boolean",
                },
            },
            followRedirects: {
                serializedName: "FollowRedirects",
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
exports.HeaderField = {
    type: {
        name: "Composite",
        className: "HeaderField",
        modelProperties: {
            headerFieldName: {
                serializedName: "key",
                type: {
                    name: "String",
                },
            },
            headerFieldValue: {
                serializedName: "value",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WebTestPropertiesValidationRules = {
    type: {
        name: "Composite",
        className: "WebTestPropertiesValidationRules",
        modelProperties: {
            contentValidation: {
                serializedName: "ContentValidation",
                type: {
                    name: "Composite",
                    className: "WebTestPropertiesValidationRulesContentValidation",
                },
            },
            sSLCheck: {
                serializedName: "SSLCheck",
                type: {
                    name: "Boolean",
                },
            },
            sSLCertRemainingLifetimeCheck: {
                serializedName: "SSLCertRemainingLifetimeCheck",
                type: {
                    name: "Number",
                },
            },
            expectedHttpStatusCode: {
                serializedName: "ExpectedHttpStatusCode",
                type: {
                    name: "Number",
                },
            },
            ignoreHttpStatusCode: {
                serializedName: "IgnoreHttpStatusCode",
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
exports.WebTestPropertiesValidationRulesContentValidation = {
    type: {
        name: "Composite",
        className: "WebTestPropertiesValidationRulesContentValidation",
        modelProperties: {
            contentMatch: {
                serializedName: "ContentMatch",
                type: {
                    name: "String",
                },
            },
            ignoreCase: {
                serializedName: "IgnoreCase",
                type: {
                    name: "Boolean",
                },
            },
            passIfTextFound: {
                serializedName: "PassIfTextFound",
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
exports.WebtestsResource = {
    type: {
        name: "Composite",
        className: "WebtestsResource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String",
                },
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentAnalyticsItem = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentAnalyticsItem",
        modelProperties: {
            id: {
                serializedName: "Id",
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "Name",
                type: {
                    name: "String",
                },
            },
            content: {
                serializedName: "Content",
                type: {
                    name: "String",
                },
            },
            version: {
                serializedName: "Version",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            scope: {
                serializedName: "Scope",
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "Type",
                type: {
                    name: "String",
                },
            },
            timeCreated: {
                serializedName: "TimeCreated",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            timeModified: {
                serializedName: "TimeModified",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            properties: {
                serializedName: "Properties",
                type: {
                    name: "Composite",
                    className: "ApplicationInsightsComponentAnalyticsItemProperties",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponentAnalyticsItemProperties = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponentAnalyticsItemProperties",
        modelProperties: {
            functionAlias: {
                serializedName: "functionAlias",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.OperationListResult = {
    type: {
        name: "Composite",
        className: "OperationListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Operation",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.Operation = {
    type: {
        name: "Composite",
        className: "Operation",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationDisplay",
                },
            },
        },
    },
};
exports.OperationDisplay = {
    type: {
        name: "Composite",
        className: "OperationDisplay",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String",
                },
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String",
                },
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String",
                },
            },
            details: {
                serializedName: "details",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorFieldContract",
                        },
                    },
                },
            },
        },
    },
};
exports.ErrorFieldContract = {
    type: {
        name: "Composite",
        className: "ErrorFieldContract",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String",
                },
            },
            target: {
                serializedName: "target",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WorkbookTemplatesListResult = {
    type: {
        name: "Composite",
        className: "WorkbookTemplatesListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WorkbookTemplate",
                        },
                    },
                },
            },
        },
    },
};
exports.WorkbookTemplateGallery = {
    type: {
        name: "Composite",
        className: "WorkbookTemplateGallery",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            category: {
                serializedName: "category",
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String",
                },
            },
            order: {
                serializedName: "order",
                type: {
                    name: "Number",
                },
            },
            resourceType: {
                serializedName: "resourceType",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WorkbookTemplateLocalizedGallery = {
    type: {
        name: "Composite",
        className: "WorkbookTemplateLocalizedGallery",
        modelProperties: {
            templateData: {
                serializedName: "templateData",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } },
                },
            },
            galleries: {
                serializedName: "galleries",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WorkbookTemplateGallery",
                        },
                    },
                },
            },
        },
    },
};
exports.WorkbookTemplateResource = {
    type: {
        name: "Composite",
        className: "WorkbookTemplateResource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String",
                },
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
        },
    },
};
exports.WorkbookTemplateError = {
    type: {
        name: "Composite",
        className: "WorkbookTemplateError",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "WorkbookTemplateErrorBody",
                },
            },
        },
    },
};
exports.WorkbookTemplateErrorBody = {
    type: {
        name: "Composite",
        className: "WorkbookTemplateErrorBody",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String",
                },
            },
            details: {
                serializedName: "details",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WorkbookTemplateErrorFieldContract",
                        },
                    },
                },
            },
        },
    },
};
exports.WorkbookTemplateErrorFieldContract = {
    type: {
        name: "Composite",
        className: "WorkbookTemplateErrorFieldContract",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String",
                },
            },
            target: {
                serializedName: "target",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WorkbookTemplateUpdateParameters = {
    type: {
        name: "Composite",
        className: "WorkbookTemplateUpdateParameters",
        modelProperties: {
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
            priority: {
                serializedName: "properties.priority",
                type: {
                    name: "Number",
                },
            },
            author: {
                serializedName: "properties.author",
                type: {
                    name: "String",
                },
            },
            templateData: {
                serializedName: "properties.templateData",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } },
                },
            },
            galleries: {
                serializedName: "properties.galleries",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WorkbookTemplateGallery",
                        },
                    },
                },
            },
            localized: {
                serializedName: "properties.localized",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "Sequence",
                            element: {
                                type: {
                                    name: "Composite",
                                    className: "WorkbookTemplateLocalizedGallery",
                                },
                            },
                        },
                    },
                },
            },
        },
    },
};
exports.WorkbooksListResult = {
    type: {
        name: "Composite",
        className: "WorkbooksListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Workbook",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.SystemData = {
    type: {
        name: "Composite",
        className: "SystemData",
        modelProperties: {
            createdBy: {
                serializedName: "createdBy",
                type: {
                    name: "String",
                },
            },
            createdByType: {
                serializedName: "createdByType",
                type: {
                    name: "String",
                },
            },
            createdAt: {
                serializedName: "createdAt",
                type: {
                    name: "DateTime",
                },
            },
            lastModifiedBy: {
                serializedName: "lastModifiedBy",
                type: {
                    name: "String",
                },
            },
            lastModifiedByType: {
                serializedName: "lastModifiedByType",
                type: {
                    name: "String",
                },
            },
            lastModifiedAt: {
                serializedName: "lastModifiedAt",
                type: {
                    name: "DateTime",
                },
            },
        },
    },
};
exports.ManagedServiceIdentity = {
    type: {
        name: "Composite",
        className: "ManagedServiceIdentity",
        modelProperties: {
            principalId: {
                serializedName: "principalId",
                readOnly: true,
                type: {
                    name: "Uuid",
                },
            },
            tenantId: {
                serializedName: "tenantId",
                readOnly: true,
                type: {
                    name: "Uuid",
                },
            },
            type: {
                serializedName: "type",
                required: true,
                type: {
                    name: "String",
                },
            },
            userAssignedIdentities: {
                serializedName: "userAssignedIdentities",
                type: {
                    name: "Dictionary",
                    value: {
                        type: { name: "Composite", className: "UserAssignedIdentity" },
                    },
                },
            },
        },
    },
};
exports.UserAssignedIdentity = {
    type: {
        name: "Composite",
        className: "UserAssignedIdentity",
        modelProperties: {
            principalId: {
                serializedName: "principalId",
                readOnly: true,
                type: {
                    name: "Uuid",
                },
            },
            clientId: {
                serializedName: "clientId",
                readOnly: true,
                type: {
                    name: "Uuid",
                },
            },
        },
    },
};
exports.Resource = {
    type: {
        name: "Composite",
        className: "Resource",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.WorkbookError = {
    type: {
        name: "Composite",
        className: "WorkbookError",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "WorkbookErrorDefinition",
                },
            },
        },
    },
};
exports.WorkbookErrorDefinition = {
    type: {
        name: "Composite",
        className: "WorkbookErrorDefinition",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            innererror: {
                serializedName: "innererror",
                type: {
                    name: "Composite",
                    className: "WorkbookInnerErrorTrace",
                },
            },
        },
    },
};
exports.WorkbookInnerErrorTrace = {
    type: {
        name: "Composite",
        className: "WorkbookInnerErrorTrace",
        modelProperties: {
            trace: {
                serializedName: "trace",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
exports.WorkbookUpdateParameters = {
    type: {
        name: "Composite",
        className: "WorkbookUpdateParameters",
        modelProperties: {
            kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            },
            tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
            displayName: {
                serializedName: "properties.displayName",
                type: {
                    name: "String",
                },
            },
            serializedData: {
                serializedName: "properties.serializedData",
                type: {
                    name: "String",
                },
            },
            category: {
                serializedName: "properties.category",
                type: {
                    name: "String",
                },
            },
            tagsPropertiesTags: {
                serializedName: "properties.tags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
            description: {
                serializedName: "properties.description",
                nullable: true,
                type: {
                    name: "String",
                },
            },
            revision: {
                serializedName: "properties.revision",
                nullable: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.LiveTokenResponse = {
    type: {
        name: "Composite",
        className: "LiveTokenResponse",
        modelProperties: {
            liveToken: {
                serializedName: "liveToken",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ErrorResponseLinkedStorage = {
    type: {
        name: "Composite",
        className: "ErrorResponseLinkedStorage",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorResponseLinkedStorageError",
                },
            },
        },
    },
};
exports.ErrorResponseLinkedStorageError = {
    type: {
        name: "Composite",
        className: "ErrorResponseLinkedStorageError",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ComponentLinkedStorageAccountsPatch = {
    type: {
        name: "Composite",
        className: "ComponentLinkedStorageAccountsPatch",
        modelProperties: {
            linkedStorageAccount: {
                serializedName: "properties.linkedStorageAccount",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.DeletedWorkbooksListResult = {
    type: {
        name: "Composite",
        className: "DeletedWorkbooksListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DeletedWorkbook",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ResourceAutoGenerated = {
    type: {
        name: "Composite",
        className: "ResourceAutoGenerated",
        modelProperties: {
            id: {
                serializedName: "id",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            name: {
                serializedName: "name",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData",
                },
            },
        },
    },
};
exports.DeletedWorkbookError = {
    type: {
        name: "Composite",
        className: "DeletedWorkbookError",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "DeletedWorkbookErrorDefinition",
                },
            },
        },
    },
};
exports.DeletedWorkbookErrorDefinition = {
    type: {
        name: "Composite",
        className: "DeletedWorkbookErrorDefinition",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            innererror: {
                serializedName: "innererror",
                type: {
                    name: "Composite",
                    className: "DeletedWorkbookInnerErrorTrace",
                },
            },
        },
    },
};
exports.DeletedWorkbookInnerErrorTrace = {
    type: {
        name: "Composite",
        className: "DeletedWorkbookInnerErrorTrace",
        modelProperties: {
            trace: {
                serializedName: "trace",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            },
        },
    },
};
exports.OperationsListResult = {
    type: {
        name: "Composite",
        className: "OperationsListResult",
        modelProperties: {
            value: {
                serializedName: "value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "OperationLive",
                        },
                    },
                },
            },
            nextLink: {
                serializedName: "nextLink",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.OperationLive = {
    type: {
        name: "Composite",
        className: "OperationLive",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String",
                },
            },
            isDataAction: {
                serializedName: "isDataAction",
                type: {
                    name: "Boolean",
                },
            },
            display: {
                serializedName: "display",
                type: {
                    name: "Composite",
                    className: "OperationInfo",
                },
            },
            origin: {
                serializedName: "origin",
                type: {
                    name: "String",
                },
            },
            properties: {
                serializedName: "properties",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } },
                },
            },
        },
    },
};
exports.OperationInfo = {
    type: {
        name: "Composite",
        className: "OperationInfo",
        modelProperties: {
            provider: {
                serializedName: "provider",
                type: {
                    name: "String",
                },
            },
            resource: {
                serializedName: "resource",
                type: {
                    name: "String",
                },
            },
            operation: {
                serializedName: "operation",
                type: {
                    name: "String",
                },
            },
            description: {
                serializedName: "description",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.ApplicationInsightsComponent = {
    type: {
        name: "Composite",
        className: "ApplicationInsightsComponent",
        modelProperties: Object.assign(Object.assign({}, exports.ComponentsResource.type.modelProperties), { kind: {
                serializedName: "kind",
                required: true,
                type: {
                    name: "String",
                },
            }, etag: {
                serializedName: "etag",
                type: {
                    name: "String",
                },
            }, applicationId: {
                serializedName: "properties.ApplicationId",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, appId: {
                serializedName: "properties.AppId",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, namePropertiesName: {
                serializedName: "properties.Name",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, applicationType: {
                defaultValue: "web",
                serializedName: "properties.Application_Type",
                type: {
                    name: "String",
                },
            }, flowType: {
                defaultValue: "Bluefield",
                serializedName: "properties.Flow_Type",
                type: {
                    name: "String",
                },
            }, requestSource: {
                defaultValue: "rest",
                serializedName: "properties.Request_Source",
                type: {
                    name: "String",
                },
            }, instrumentationKey: {
                serializedName: "properties.InstrumentationKey",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, creationDate: {
                serializedName: "properties.CreationDate",
                readOnly: true,
                type: {
                    name: "DateTime",
                },
            }, tenantId: {
                serializedName: "properties.TenantId",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, hockeyAppId: {
                serializedName: "properties.HockeyAppId",
                type: {
                    name: "String",
                },
            }, hockeyAppToken: {
                serializedName: "properties.HockeyAppToken",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, samplingPercentage: {
                serializedName: "properties.SamplingPercentage",
                type: {
                    name: "Number",
                },
            }, connectionString: {
                serializedName: "properties.ConnectionString",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, retentionInDays: {
                serializedName: "properties.RetentionInDays",
                type: {
                    name: "Number",
                },
            }, disableIpMasking: {
                serializedName: "properties.DisableIpMasking",
                type: {
                    name: "Boolean",
                },
            }, immediatePurgeDataOn30Days: {
                serializedName: "properties.ImmediatePurgeDataOn30Days",
                type: {
                    name: "Boolean",
                },
            }, workspaceResourceId: {
                serializedName: "properties.WorkspaceResourceId",
                type: {
                    name: "String",
                },
            }, laMigrationDate: {
                serializedName: "properties.LaMigrationDate",
                readOnly: true,
                type: {
                    name: "DateTime",
                },
            }, privateLinkScopedResources: {
                serializedName: "properties.PrivateLinkScopedResources",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "PrivateLinkScopedResource",
                        },
                    },
                },
            }, publicNetworkAccessForIngestion: {
                defaultValue: "Enabled",
                serializedName: "properties.publicNetworkAccessForIngestion",
                type: {
                    name: "String",
                },
            }, publicNetworkAccessForQuery: {
                defaultValue: "Enabled",
                serializedName: "properties.publicNetworkAccessForQuery",
                type: {
                    name: "String",
                },
            }, ingestionMode: {
                defaultValue: "LogAnalytics",
                serializedName: "properties.IngestionMode",
                type: {
                    name: "String",
                },
            }, disableLocalAuth: {
                serializedName: "properties.DisableLocalAuth",
                type: {
                    name: "Boolean",
                },
            }, forceCustomerStorageForProfiler: {
                serializedName: "properties.ForceCustomerStorageForProfiler",
                type: {
                    name: "Boolean",
                },
            } }),
    },
};
exports.WebTest = {
    type: {
        name: "Composite",
        className: "WebTest",
        modelProperties: Object.assign(Object.assign({}, exports.WebtestsResource.type.modelProperties), { kind: {
                serializedName: "kind",
                type: {
                    name: "Enum",
                    allowedValues: ["ping", "multistep", "standard"],
                },
            }, syntheticMonitorId: {
                serializedName: "properties.SyntheticMonitorId",
                type: {
                    name: "String",
                },
            }, webTestName: {
                serializedName: "properties.Name",
                type: {
                    name: "String",
                },
            }, description: {
                serializedName: "properties.Description",
                type: {
                    name: "String",
                },
            }, enabled: {
                serializedName: "properties.Enabled",
                type: {
                    name: "Boolean",
                },
            }, frequency: {
                defaultValue: 300,
                serializedName: "properties.Frequency",
                type: {
                    name: "Number",
                },
            }, timeout: {
                defaultValue: 30,
                serializedName: "properties.Timeout",
                type: {
                    name: "Number",
                },
            }, webTestKind: {
                serializedName: "properties.Kind",
                type: {
                    name: "Enum",
                    allowedValues: ["ping", "multistep", "standard"],
                },
            }, retryEnabled: {
                serializedName: "properties.RetryEnabled",
                type: {
                    name: "Boolean",
                },
            }, locations: {
                serializedName: "properties.Locations",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WebTestGeolocation",
                        },
                    },
                },
            }, configuration: {
                serializedName: "properties.Configuration",
                type: {
                    name: "Composite",
                    className: "WebTestPropertiesConfiguration",
                },
            }, provisioningState: {
                serializedName: "properties.provisioningState",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, request: {
                serializedName: "properties.Request",
                type: {
                    name: "Composite",
                    className: "WebTestPropertiesRequest",
                },
            }, validationRules: {
                serializedName: "properties.ValidationRules",
                type: {
                    name: "Composite",
                    className: "WebTestPropertiesValidationRules",
                },
            } }),
    },
};
exports.WorkbookTemplate = {
    type: {
        name: "Composite",
        className: "WorkbookTemplate",
        modelProperties: Object.assign(Object.assign({}, exports.WorkbookTemplateResource.type.modelProperties), { priority: {
                serializedName: "properties.priority",
                type: {
                    name: "Number",
                },
            }, author: {
                serializedName: "properties.author",
                type: {
                    name: "String",
                },
            }, templateData: {
                serializedName: "properties.templateData",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } },
                },
            }, galleries: {
                serializedName: "properties.galleries",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WorkbookTemplateGallery",
                        },
                    },
                },
            }, localized: {
                serializedName: "properties.localized",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "Sequence",
                            element: {
                                type: {
                                    name: "Composite",
                                    className: "WorkbookTemplateLocalizedGallery",
                                },
                            },
                        },
                    },
                },
            } }),
    },
};
exports.WorkbookResourceIdentity = {
    type: {
        name: "Composite",
        className: "WorkbookResourceIdentity",
        modelProperties: Object.assign({}, exports.ManagedServiceIdentity.type.modelProperties),
    },
};
exports.TrackedResource = {
    type: {
        name: "Composite",
        className: "TrackedResource",
        modelProperties: Object.assign(Object.assign({}, exports.Resource.type.modelProperties), { tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            }, location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String",
                },
            } }),
    },
};
exports.ProxyResource = {
    type: {
        name: "Composite",
        className: "ProxyResource",
        modelProperties: Object.assign({}, exports.Resource.type.modelProperties),
    },
};
exports.TrackedResourceAutoGenerated = {
    type: {
        name: "Composite",
        className: "TrackedResourceAutoGenerated",
        modelProperties: Object.assign(Object.assign({}, exports.ResourceAutoGenerated.type.modelProperties), { tags: {
                serializedName: "tags",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            }, location: {
                serializedName: "location",
                required: true,
                type: {
                    name: "String",
                },
            } }),
    },
};
exports.WorkbookResource = {
    type: {
        name: "Composite",
        className: "WorkbookResource",
        modelProperties: Object.assign(Object.assign({}, exports.TrackedResource.type.modelProperties), { identity: {
                serializedName: "identity",
                type: {
                    name: "Composite",
                    className: "WorkbookResourceIdentity",
                },
            }, kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            }, etag: {
                serializedName: "etag",
                type: {
                    name: "String",
                },
            } }),
    },
};
exports.ComponentLinkedStorageAccounts = {
    type: {
        name: "Composite",
        className: "ComponentLinkedStorageAccounts",
        modelProperties: Object.assign(Object.assign({}, exports.ProxyResource.type.modelProperties), { linkedStorageAccount: {
                serializedName: "properties.linkedStorageAccount",
                type: {
                    name: "String",
                },
            } }),
    },
};
exports.DeletedWorkbookResource = {
    type: {
        name: "Composite",
        className: "DeletedWorkbookResource",
        modelProperties: Object.assign(Object.assign({}, exports.TrackedResourceAutoGenerated.type.modelProperties), { kind: {
                serializedName: "kind",
                type: {
                    name: "String",
                },
            }, etag: {
                serializedName: "etag",
                type: {
                    name: "String",
                },
            } }),
    },
};
exports.Workbook = {
    type: {
        name: "Composite",
        className: "Workbook",
        modelProperties: Object.assign(Object.assign({}, exports.WorkbookResource.type.modelProperties), { systemData: {
                serializedName: "systemData",
                type: {
                    name: "Composite",
                    className: "SystemData",
                },
            }, displayName: {
                serializedName: "properties.displayName",
                type: {
                    name: "String",
                },
            }, serializedData: {
                serializedName: "properties.serializedData",
                nullable: true,
                type: {
                    name: "String",
                },
            }, version: {
                serializedName: "properties.version",
                type: {
                    name: "String",
                },
            }, timeModified: {
                serializedName: "properties.timeModified",
                readOnly: true,
                type: {
                    name: "DateTime",
                },
            }, category: {
                serializedName: "properties.category",
                type: {
                    name: "String",
                },
            }, tagsPropertiesTags: {
                serializedName: "properties.tags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            }, userId: {
                serializedName: "properties.userId",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, sourceId: {
                serializedName: "properties.sourceId",
                type: {
                    name: "String",
                },
            }, storageUri: {
                serializedName: "properties.storageUri",
                nullable: true,
                type: {
                    name: "String",
                },
            }, description: {
                serializedName: "properties.description",
                nullable: true,
                type: {
                    name: "String",
                },
            }, revision: {
                serializedName: "properties.revision",
                readOnly: true,
                nullable: true,
                type: {
                    name: "String",
                },
            } }),
    },
};
exports.DeletedWorkbook = {
    type: {
        name: "Composite",
        className: "DeletedWorkbook",
        modelProperties: Object.assign(Object.assign({}, exports.DeletedWorkbookResource.type.modelProperties), { displayName: {
                serializedName: "properties.displayName",
                type: {
                    name: "String",
                },
            }, serializedData: {
                serializedName: "properties.serializedData",
                nullable: true,
                type: {
                    name: "String",
                },
            }, version: {
                serializedName: "properties.version",
                type: {
                    name: "String",
                },
            }, timeModified: {
                serializedName: "properties.timeModified",
                readOnly: true,
                type: {
                    name: "DateTime",
                },
            }, category: {
                serializedName: "properties.category",
                type: {
                    name: "String",
                },
            }, tagsPropertiesTags: {
                serializedName: "properties.tags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String",
                        },
                    },
                },
            }, userId: {
                serializedName: "properties.userId",
                readOnly: true,
                type: {
                    name: "String",
                },
            }, sourceId: {
                serializedName: "properties.sourceId",
                type: {
                    name: "String",
                },
            }, storageUri: {
                serializedName: "properties.storageUri",
                nullable: true,
                type: {
                    name: "String",
                },
            }, description: {
                serializedName: "properties.description",
                nullable: true,
                type: {
                    name: "String",
                },
            }, revision: {
                serializedName: "properties.revision",
                readOnly: true,
                nullable: true,
                type: {
                    name: "String",
                },
            } }),
    },
};
