// @azure/arm-costmanagement/dist/commonjs/models/index.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnownCheckNameAvailabilityReason = exports.KnownCreatedByType = exports.KnownScheduledActionKind = exports.KnownScheduledActionStatus = exports.KnownWeeksOfMonth = exports.KnownDaysOfWeek = exports.KnownScheduleFrequency = exports.KnownFileFormat = exports.KnownReportOperationStatusType = exports.KnownGenerateDetailedCostReportMetricType = exports.KnownCostDetailsDataFormat = exports.KnownCostDetailsStatusType = exports.KnownCostDetailsMetricType = exports.KnownExecutionStatus = exports.KnownExecutionType = exports.KnownFormatType = exports.KnownRecurrenceType = exports.KnownStatusType = exports.KnownReservationReportSchema = exports.KnownOperationStatusType = exports.KnownQueryOperatorType = exports.KnownTimeframeType = exports.KnownExportType = exports.KnownForecastOperatorType = exports.KnownFunctionName = exports.KnownGranularityType = exports.KnownForecastTimeframe = exports.KnownForecastType = exports.KnownExternalCloudProviderType = exports.KnownAlertStatus = exports.KnownAlertOperator = exports.KnownAlertTimeGrainType = exports.KnownAlertSource = exports.KnownAlertCriteria = exports.KnownAlertCategory = exports.KnownAlertType = exports.KnownPivotType = exports.KnownKpiType = exports.KnownMetricType = exports.KnownAccumulatedType = exports.KnownChartType = exports.KnownOperatorType = exports.KnownReportConfigSortingType = exports.KnownQueryColumnType = exports.KnownFunctionType = exports.KnownReportGranularityType = exports.KnownReportTimeframeType = exports.KnownReportType = exports.KnownActionType = exports.KnownOrigin = void 0;
exports.KnownGrainParameter = exports.KnownBenefitKind = exports.KnownScope = exports.KnownTerm = exports.KnownGrain = exports.KnownLookBackPeriod = void 0;
/** Known values of {@link Origin} that the service accepts. */
var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (exports.KnownOrigin = KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (exports.KnownActionType = KnownActionType = {}));
/** Known values of {@link ReportType} that the service accepts. */
var KnownReportType;
(function (KnownReportType) {
    /** Usage */
    KnownReportType["Usage"] = "Usage";
})(KnownReportType || (exports.KnownReportType = KnownReportType = {}));
/** Known values of {@link ReportTimeframeType} that the service accepts. */
var KnownReportTimeframeType;
(function (KnownReportTimeframeType) {
    /** WeekToDate */
    KnownReportTimeframeType["WeekToDate"] = "WeekToDate";
    /** MonthToDate */
    KnownReportTimeframeType["MonthToDate"] = "MonthToDate";
    /** YearToDate */
    KnownReportTimeframeType["YearToDate"] = "YearToDate";
    /** Custom */
    KnownReportTimeframeType["Custom"] = "Custom";
})(KnownReportTimeframeType || (exports.KnownReportTimeframeType = KnownReportTimeframeType = {}));
/** Known values of {@link ReportGranularityType} that the service accepts. */
var KnownReportGranularityType;
(function (KnownReportGranularityType) {
    /** Daily */
    KnownReportGranularityType["Daily"] = "Daily";
    /** Monthly */
    KnownReportGranularityType["Monthly"] = "Monthly";
})(KnownReportGranularityType || (exports.KnownReportGranularityType = KnownReportGranularityType = {}));
/** Known values of {@link FunctionType} that the service accepts. */
var KnownFunctionType;
(function (KnownFunctionType) {
    /** Sum */
    KnownFunctionType["Sum"] = "Sum";
})(KnownFunctionType || (exports.KnownFunctionType = KnownFunctionType = {}));
/** Known values of {@link QueryColumnType} that the service accepts. */
var KnownQueryColumnType;
(function (KnownQueryColumnType) {
    /** The tag associated with the cost data. */
    KnownQueryColumnType["TagKey"] = "TagKey";
    /** The dimension of cost data. */
    KnownQueryColumnType["Dimension"] = "Dimension";
})(KnownQueryColumnType || (exports.KnownQueryColumnType = KnownQueryColumnType = {}));
/** Known values of {@link ReportConfigSortingType} that the service accepts. */
var KnownReportConfigSortingType;
(function (KnownReportConfigSortingType) {
    /** Ascending */
    KnownReportConfigSortingType["Ascending"] = "Ascending";
    /** Descending */
    KnownReportConfigSortingType["Descending"] = "Descending";
})(KnownReportConfigSortingType || (exports.KnownReportConfigSortingType = KnownReportConfigSortingType = {}));
/** Known values of {@link OperatorType} that the service accepts. */
var KnownOperatorType;
(function (KnownOperatorType) {
    /** In */
    KnownOperatorType["In"] = "In";
    /** Contains */
    KnownOperatorType["Contains"] = "Contains";
})(KnownOperatorType || (exports.KnownOperatorType = KnownOperatorType = {}));
/** Known values of {@link ChartType} that the service accepts. */
var KnownChartType;
(function (KnownChartType) {
    /** Area */
    KnownChartType["Area"] = "Area";
    /** Line */
    KnownChartType["Line"] = "Line";
    /** StackedColumn */
    KnownChartType["StackedColumn"] = "StackedColumn";
    /** GroupedColumn */
    KnownChartType["GroupedColumn"] = "GroupedColumn";
    /** Table */
    KnownChartType["Table"] = "Table";
})(KnownChartType || (exports.KnownChartType = KnownChartType = {}));
/** Known values of {@link AccumulatedType} that the service accepts. */
var KnownAccumulatedType;
(function (KnownAccumulatedType) {
    /** True */
    KnownAccumulatedType["True"] = "true";
    /** False */
    KnownAccumulatedType["False"] = "false";
})(KnownAccumulatedType || (exports.KnownAccumulatedType = KnownAccumulatedType = {}));
/** Known values of {@link MetricType} that the service accepts. */
var KnownMetricType;
(function (KnownMetricType) {
    /** ActualCost */
    KnownMetricType["ActualCost"] = "ActualCost";
    /** AmortizedCost */
    KnownMetricType["AmortizedCost"] = "AmortizedCost";
    /** Ahub */
    KnownMetricType["Ahub"] = "AHUB";
})(KnownMetricType || (exports.KnownMetricType = KnownMetricType = {}));
/** Known values of {@link KpiType} that the service accepts. */
var KnownKpiType;
(function (KnownKpiType) {
    /** Forecast */
    KnownKpiType["Forecast"] = "Forecast";
    /** Budget */
    KnownKpiType["Budget"] = "Budget";
})(KnownKpiType || (exports.KnownKpiType = KnownKpiType = {}));
/** Known values of {@link PivotType} that the service accepts. */
var KnownPivotType;
(function (KnownPivotType) {
    /** Dimension */
    KnownPivotType["Dimension"] = "Dimension";
    /** TagKey */
    KnownPivotType["TagKey"] = "TagKey";
})(KnownPivotType || (exports.KnownPivotType = KnownPivotType = {}));
/** Known values of {@link AlertType} that the service accepts. */
var KnownAlertType;
(function (KnownAlertType) {
    /** Budget */
    KnownAlertType["Budget"] = "Budget";
    /** Invoice */
    KnownAlertType["Invoice"] = "Invoice";
    /** Credit */
    KnownAlertType["Credit"] = "Credit";
    /** Quota */
    KnownAlertType["Quota"] = "Quota";
    /** General */
    KnownAlertType["General"] = "General";
    /** XCloud */
    KnownAlertType["XCloud"] = "xCloud";
    /** BudgetForecast */
    KnownAlertType["BudgetForecast"] = "BudgetForecast";
})(KnownAlertType || (exports.KnownAlertType = KnownAlertType = {}));
/** Known values of {@link AlertCategory} that the service accepts. */
var KnownAlertCategory;
(function (KnownAlertCategory) {
    /** Cost */
    KnownAlertCategory["Cost"] = "Cost";
    /** Usage */
    KnownAlertCategory["Usage"] = "Usage";
    /** Billing */
    KnownAlertCategory["Billing"] = "Billing";
    /** System */
    KnownAlertCategory["System"] = "System";
})(KnownAlertCategory || (exports.KnownAlertCategory = KnownAlertCategory = {}));
/** Known values of {@link AlertCriteria} that the service accepts. */
var KnownAlertCriteria;
(function (KnownAlertCriteria) {
    /** CostThresholdExceeded */
    KnownAlertCriteria["CostThresholdExceeded"] = "CostThresholdExceeded";
    /** UsageThresholdExceeded */
    KnownAlertCriteria["UsageThresholdExceeded"] = "UsageThresholdExceeded";
    /** CreditThresholdApproaching */
    KnownAlertCriteria["CreditThresholdApproaching"] = "CreditThresholdApproaching";
    /** CreditThresholdReached */
    KnownAlertCriteria["CreditThresholdReached"] = "CreditThresholdReached";
    /** QuotaThresholdApproaching */
    KnownAlertCriteria["QuotaThresholdApproaching"] = "QuotaThresholdApproaching";
    /** QuotaThresholdReached */
    KnownAlertCriteria["QuotaThresholdReached"] = "QuotaThresholdReached";
    /** MultiCurrency */
    KnownAlertCriteria["MultiCurrency"] = "MultiCurrency";
    /** ForecastCostThresholdExceeded */
    KnownAlertCriteria["ForecastCostThresholdExceeded"] = "ForecastCostThresholdExceeded";
    /** ForecastUsageThresholdExceeded */
    KnownAlertCriteria["ForecastUsageThresholdExceeded"] = "ForecastUsageThresholdExceeded";
    /** InvoiceDueDateApproaching */
    KnownAlertCriteria["InvoiceDueDateApproaching"] = "InvoiceDueDateApproaching";
    /** InvoiceDueDateReached */
    KnownAlertCriteria["InvoiceDueDateReached"] = "InvoiceDueDateReached";
    /** CrossCloudNewDataAvailable */
    KnownAlertCriteria["CrossCloudNewDataAvailable"] = "CrossCloudNewDataAvailable";
    /** CrossCloudCollectionError */
    KnownAlertCriteria["CrossCloudCollectionError"] = "CrossCloudCollectionError";
    /** GeneralThresholdError */
    KnownAlertCriteria["GeneralThresholdError"] = "GeneralThresholdError";
})(KnownAlertCriteria || (exports.KnownAlertCriteria = KnownAlertCriteria = {}));
/** Known values of {@link AlertSource} that the service accepts. */
var KnownAlertSource;
(function (KnownAlertSource) {
    /** Preset */
    KnownAlertSource["Preset"] = "Preset";
    /** User */
    KnownAlertSource["User"] = "User";
})(KnownAlertSource || (exports.KnownAlertSource = KnownAlertSource = {}));
/** Known values of {@link AlertTimeGrainType} that the service accepts. */
var KnownAlertTimeGrainType;
(function (KnownAlertTimeGrainType) {
    /** None */
    KnownAlertTimeGrainType["None"] = "None";
    /** Monthly */
    KnownAlertTimeGrainType["Monthly"] = "Monthly";
    /** Quarterly */
    KnownAlertTimeGrainType["Quarterly"] = "Quarterly";
    /** Annually */
    KnownAlertTimeGrainType["Annually"] = "Annually";
    /** BillingMonth */
    KnownAlertTimeGrainType["BillingMonth"] = "BillingMonth";
    /** BillingQuarter */
    KnownAlertTimeGrainType["BillingQuarter"] = "BillingQuarter";
    /** BillingAnnual */
    KnownAlertTimeGrainType["BillingAnnual"] = "BillingAnnual";
})(KnownAlertTimeGrainType || (exports.KnownAlertTimeGrainType = KnownAlertTimeGrainType = {}));
/** Known values of {@link AlertOperator} that the service accepts. */
var KnownAlertOperator;
(function (KnownAlertOperator) {
    /** None */
    KnownAlertOperator["None"] = "None";
    /** EqualTo */
    KnownAlertOperator["EqualTo"] = "EqualTo";
    /** GreaterThan */
    KnownAlertOperator["GreaterThan"] = "GreaterThan";
    /** GreaterThanOrEqualTo */
    KnownAlertOperator["GreaterThanOrEqualTo"] = "GreaterThanOrEqualTo";
    /** LessThan */
    KnownAlertOperator["LessThan"] = "LessThan";
    /** LessThanOrEqualTo */
    KnownAlertOperator["LessThanOrEqualTo"] = "LessThanOrEqualTo";
})(KnownAlertOperator || (exports.KnownAlertOperator = KnownAlertOperator = {}));
/** Known values of {@link AlertStatus} that the service accepts. */
var KnownAlertStatus;
(function (KnownAlertStatus) {
    /** None */
    KnownAlertStatus["None"] = "None";
    /** Active */
    KnownAlertStatus["Active"] = "Active";
    /** Overridden */
    KnownAlertStatus["Overridden"] = "Overridden";
    /** Resolved */
    KnownAlertStatus["Resolved"] = "Resolved";
    /** Dismissed */
    KnownAlertStatus["Dismissed"] = "Dismissed";
})(KnownAlertStatus || (exports.KnownAlertStatus = KnownAlertStatus = {}));
/** Known values of {@link ExternalCloudProviderType} that the service accepts. */
var KnownExternalCloudProviderType;
(function (KnownExternalCloudProviderType) {
    /** ExternalSubscriptions */
    KnownExternalCloudProviderType["ExternalSubscriptions"] = "externalSubscriptions";
    /** ExternalBillingAccounts */
    KnownExternalCloudProviderType["ExternalBillingAccounts"] = "externalBillingAccounts";
})(KnownExternalCloudProviderType || (exports.KnownExternalCloudProviderType = KnownExternalCloudProviderType = {}));
/** Known values of {@link ForecastType} that the service accepts. */
var KnownForecastType;
(function (KnownForecastType) {
    /** Usage */
    KnownForecastType["Usage"] = "Usage";
    /** ActualCost */
    KnownForecastType["ActualCost"] = "ActualCost";
    /** AmortizedCost */
    KnownForecastType["AmortizedCost"] = "AmortizedCost";
})(KnownForecastType || (exports.KnownForecastType = KnownForecastType = {}));
/** Known values of {@link ForecastTimeframe} that the service accepts. */
var KnownForecastTimeframe;
(function (KnownForecastTimeframe) {
    /** Custom */
    KnownForecastTimeframe["Custom"] = "Custom";
})(KnownForecastTimeframe || (exports.KnownForecastTimeframe = KnownForecastTimeframe = {}));
/** Known values of {@link GranularityType} that the service accepts. */
var KnownGranularityType;
(function (KnownGranularityType) {
    /** Daily */
    KnownGranularityType["Daily"] = "Daily";
})(KnownGranularityType || (exports.KnownGranularityType = KnownGranularityType = {}));
/** Known values of {@link FunctionName} that the service accepts. */
var KnownFunctionName;
(function (KnownFunctionName) {
    /** PreTaxCostUSD */
    KnownFunctionName["PreTaxCostUSD"] = "PreTaxCostUSD";
    /** Cost */
    KnownFunctionName["Cost"] = "Cost";
    /** CostUSD */
    KnownFunctionName["CostUSD"] = "CostUSD";
    /** PreTaxCost */
    KnownFunctionName["PreTaxCost"] = "PreTaxCost";
})(KnownFunctionName || (exports.KnownFunctionName = KnownFunctionName = {}));
/** Known values of {@link ForecastOperatorType} that the service accepts. */
var KnownForecastOperatorType;
(function (KnownForecastOperatorType) {
    /** In */
    KnownForecastOperatorType["In"] = "In";
})(KnownForecastOperatorType || (exports.KnownForecastOperatorType = KnownForecastOperatorType = {}));
/** Known values of {@link ExportType} that the service accepts. */
var KnownExportType;
(function (KnownExportType) {
    /** Usage */
    KnownExportType["Usage"] = "Usage";
    /** ActualCost */
    KnownExportType["ActualCost"] = "ActualCost";
    /** AmortizedCost */
    KnownExportType["AmortizedCost"] = "AmortizedCost";
})(KnownExportType || (exports.KnownExportType = KnownExportType = {}));
/** Known values of {@link TimeframeType} that the service accepts. */
var KnownTimeframeType;
(function (KnownTimeframeType) {
    /** MonthToDate */
    KnownTimeframeType["MonthToDate"] = "MonthToDate";
    /** BillingMonthToDate */
    KnownTimeframeType["BillingMonthToDate"] = "BillingMonthToDate";
    /** TheLastMonth */
    KnownTimeframeType["TheLastMonth"] = "TheLastMonth";
    /** TheLastBillingMonth */
    KnownTimeframeType["TheLastBillingMonth"] = "TheLastBillingMonth";
    /** WeekToDate */
    KnownTimeframeType["WeekToDate"] = "WeekToDate";
    /** Custom */
    KnownTimeframeType["Custom"] = "Custom";
})(KnownTimeframeType || (exports.KnownTimeframeType = KnownTimeframeType = {}));
/** Known values of {@link QueryOperatorType} that the service accepts. */
var KnownQueryOperatorType;
(function (KnownQueryOperatorType) {
    /** In */
    KnownQueryOperatorType["In"] = "In";
})(KnownQueryOperatorType || (exports.KnownQueryOperatorType = KnownQueryOperatorType = {}));
/** Known values of {@link OperationStatusType} that the service accepts. */
var KnownOperationStatusType;
(function (KnownOperationStatusType) {
    /** Running */
    KnownOperationStatusType["Running"] = "Running";
    /** Completed */
    KnownOperationStatusType["Completed"] = "Completed";
    /** Failed */
    KnownOperationStatusType["Failed"] = "Failed";
})(KnownOperationStatusType || (exports.KnownOperationStatusType = KnownOperationStatusType = {}));
/** Known values of {@link ReservationReportSchema} that the service accepts. */
var KnownReservationReportSchema;
(function (KnownReservationReportSchema) {
    /** InstanceFlexibilityGroup */
    KnownReservationReportSchema["InstanceFlexibilityGroup"] = "InstanceFlexibilityGroup";
    /** InstanceFlexibilityRatio */
    KnownReservationReportSchema["InstanceFlexibilityRatio"] = "InstanceFlexibilityRatio";
    /** InstanceId */
    KnownReservationReportSchema["InstanceId"] = "InstanceId";
    /** Kind */
    KnownReservationReportSchema["Kind"] = "Kind";
    /** ReservationId */
    KnownReservationReportSchema["ReservationId"] = "ReservationId";
    /** ReservationOrderId */
    KnownReservationReportSchema["ReservationOrderId"] = "ReservationOrderId";
    /** ReservedHours */
    KnownReservationReportSchema["ReservedHours"] = "ReservedHours";
    /** SkuName */
    KnownReservationReportSchema["SkuName"] = "SkuName";
    /** TotalReservedQuantity */
    KnownReservationReportSchema["TotalReservedQuantity"] = "TotalReservedQuantity";
    /** UsageDate */
    KnownReservationReportSchema["UsageDate"] = "UsageDate";
    /** UsedHours */
    KnownReservationReportSchema["UsedHours"] = "UsedHours";
})(KnownReservationReportSchema || (exports.KnownReservationReportSchema = KnownReservationReportSchema = {}));
/** Known values of {@link StatusType} that the service accepts. */
var KnownStatusType;
(function (KnownStatusType) {
    /** Active */
    KnownStatusType["Active"] = "Active";
    /** Inactive */
    KnownStatusType["Inactive"] = "Inactive";
})(KnownStatusType || (exports.KnownStatusType = KnownStatusType = {}));
/** Known values of {@link RecurrenceType} that the service accepts. */
var KnownRecurrenceType;
(function (KnownRecurrenceType) {
    /** Daily */
    KnownRecurrenceType["Daily"] = "Daily";
    /** Weekly */
    KnownRecurrenceType["Weekly"] = "Weekly";
    /** Monthly */
    KnownRecurrenceType["Monthly"] = "Monthly";
    /** Annually */
    KnownRecurrenceType["Annually"] = "Annually";
})(KnownRecurrenceType || (exports.KnownRecurrenceType = KnownRecurrenceType = {}));
/** Known values of {@link FormatType} that the service accepts. */
var KnownFormatType;
(function (KnownFormatType) {
    /** Csv */
    KnownFormatType["Csv"] = "Csv";
})(KnownFormatType || (exports.KnownFormatType = KnownFormatType = {}));
/** Known values of {@link ExecutionType} that the service accepts. */
var KnownExecutionType;
(function (KnownExecutionType) {
    /** OnDemand */
    KnownExecutionType["OnDemand"] = "OnDemand";
    /** Scheduled */
    KnownExecutionType["Scheduled"] = "Scheduled";
})(KnownExecutionType || (exports.KnownExecutionType = KnownExecutionType = {}));
/** Known values of {@link ExecutionStatus} that the service accepts. */
var KnownExecutionStatus;
(function (KnownExecutionStatus) {
    /** Queued */
    KnownExecutionStatus["Queued"] = "Queued";
    /** InProgress */
    KnownExecutionStatus["InProgress"] = "InProgress";
    /** Completed */
    KnownExecutionStatus["Completed"] = "Completed";
    /** Failed */
    KnownExecutionStatus["Failed"] = "Failed";
    /** Timeout */
    KnownExecutionStatus["Timeout"] = "Timeout";
    /** NewDataNotAvailable */
    KnownExecutionStatus["NewDataNotAvailable"] = "NewDataNotAvailable";
    /** DataNotAvailable */
    KnownExecutionStatus["DataNotAvailable"] = "DataNotAvailable";
})(KnownExecutionStatus || (exports.KnownExecutionStatus = KnownExecutionStatus = {}));
/** Known values of {@link CostDetailsMetricType} that the service accepts. */
var KnownCostDetailsMetricType;
(function (KnownCostDetailsMetricType) {
    /** Actual cost data. */
    KnownCostDetailsMetricType["ActualCostCostDetailsMetricType"] = "ActualCost";
    /** Amortized cost data. */
    KnownCostDetailsMetricType["AmortizedCostCostDetailsMetricType"] = "AmortizedCost";
})(KnownCostDetailsMetricType || (exports.KnownCostDetailsMetricType = KnownCostDetailsMetricType = {}));
/** Known values of {@link CostDetailsStatusType} that the service accepts. */
var KnownCostDetailsStatusType;
(function (KnownCostDetailsStatusType) {
    /** Operation is Completed. */
    KnownCostDetailsStatusType["CompletedCostDetailsStatusType"] = "Completed";
    /** Operation is Completed and no cost data found. */
    KnownCostDetailsStatusType["NoDataFoundCostDetailsStatusType"] = "NoDataFound";
    /** Operation Failed. */
    KnownCostDetailsStatusType["FailedCostDetailsStatusType"] = "Failed";
})(KnownCostDetailsStatusType || (exports.KnownCostDetailsStatusType = KnownCostDetailsStatusType = {}));
/** Known values of {@link CostDetailsDataFormat} that the service accepts. */
var KnownCostDetailsDataFormat;
(function (KnownCostDetailsDataFormat) {
    /** Csv data format. */
    KnownCostDetailsDataFormat["CsvCostDetailsDataFormat"] = "Csv";
})(KnownCostDetailsDataFormat || (exports.KnownCostDetailsDataFormat = KnownCostDetailsDataFormat = {}));
/** Known values of {@link GenerateDetailedCostReportMetricType} that the service accepts. */
var KnownGenerateDetailedCostReportMetricType;
(function (KnownGenerateDetailedCostReportMetricType) {
    /** ActualCost */
    KnownGenerateDetailedCostReportMetricType["ActualCost"] = "ActualCost";
    /** AmortizedCost */
    KnownGenerateDetailedCostReportMetricType["AmortizedCost"] = "AmortizedCost";
})(KnownGenerateDetailedCostReportMetricType || (exports.KnownGenerateDetailedCostReportMetricType = KnownGenerateDetailedCostReportMetricType = {}));
/** Known values of {@link ReportOperationStatusType} that the service accepts. */
var KnownReportOperationStatusType;
(function (KnownReportOperationStatusType) {
    /** InProgress */
    KnownReportOperationStatusType["InProgress"] = "InProgress";
    /** Completed */
    KnownReportOperationStatusType["Completed"] = "Completed";
    /** Failed */
    KnownReportOperationStatusType["Failed"] = "Failed";
    /** Queued */
    KnownReportOperationStatusType["Queued"] = "Queued";
    /** NoDataFound */
    KnownReportOperationStatusType["NoDataFound"] = "NoDataFound";
    /** ReadyToDownload */
    KnownReportOperationStatusType["ReadyToDownload"] = "ReadyToDownload";
    /** TimedOut */
    KnownReportOperationStatusType["TimedOut"] = "TimedOut";
})(KnownReportOperationStatusType || (exports.KnownReportOperationStatusType = KnownReportOperationStatusType = {}));
/** Known values of {@link FileFormat} that the service accepts. */
var KnownFileFormat;
(function (KnownFileFormat) {
    /** Csv */
    KnownFileFormat["Csv"] = "Csv";
})(KnownFileFormat || (exports.KnownFileFormat = KnownFileFormat = {}));
/** Known values of {@link ScheduleFrequency} that the service accepts. */
var KnownScheduleFrequency;
(function (KnownScheduleFrequency) {
    /** Cost analysis data will be emailed every day. */
    KnownScheduleFrequency["Daily"] = "Daily";
    /** Cost analysis data will be emailed every week. */
    KnownScheduleFrequency["Weekly"] = "Weekly";
    /** Cost analysis data will be emailed every month. */
    KnownScheduleFrequency["Monthly"] = "Monthly";
})(KnownScheduleFrequency || (exports.KnownScheduleFrequency = KnownScheduleFrequency = {}));
/** Known values of {@link DaysOfWeek} that the service accepts. */
var KnownDaysOfWeek;
(function (KnownDaysOfWeek) {
    /** Monday */
    KnownDaysOfWeek["Monday"] = "Monday";
    /** Tuesday */
    KnownDaysOfWeek["Tuesday"] = "Tuesday";
    /** Wednesday */
    KnownDaysOfWeek["Wednesday"] = "Wednesday";
    /** Thursday */
    KnownDaysOfWeek["Thursday"] = "Thursday";
    /** Friday */
    KnownDaysOfWeek["Friday"] = "Friday";
    /** Saturday */
    KnownDaysOfWeek["Saturday"] = "Saturday";
    /** Sunday */
    KnownDaysOfWeek["Sunday"] = "Sunday";
})(KnownDaysOfWeek || (exports.KnownDaysOfWeek = KnownDaysOfWeek = {}));
/** Known values of {@link WeeksOfMonth} that the service accepts. */
var KnownWeeksOfMonth;
(function (KnownWeeksOfMonth) {
    /** First */
    KnownWeeksOfMonth["First"] = "First";
    /** Second */
    KnownWeeksOfMonth["Second"] = "Second";
    /** Third */
    KnownWeeksOfMonth["Third"] = "Third";
    /** Fourth */
    KnownWeeksOfMonth["Fourth"] = "Fourth";
    /** Last */
    KnownWeeksOfMonth["Last"] = "Last";
})(KnownWeeksOfMonth || (exports.KnownWeeksOfMonth = KnownWeeksOfMonth = {}));
/** Known values of {@link ScheduledActionStatus} that the service accepts. */
var KnownScheduledActionStatus;
(function (KnownScheduledActionStatus) {
    /** Scheduled action is saved but will not be run. */
    KnownScheduledActionStatus["Disabled"] = "Disabled";
    /** Scheduled action is saved and will be run. */
    KnownScheduledActionStatus["Enabled"] = "Enabled";
    /** Scheduled action is expired. */
    KnownScheduledActionStatus["Expired"] = "Expired";
})(KnownScheduledActionStatus || (exports.KnownScheduledActionStatus = KnownScheduledActionStatus = {}));
/** Known values of {@link ScheduledActionKind} that the service accepts. */
var KnownScheduledActionKind;
(function (KnownScheduledActionKind) {
    /** Cost analysis data will be emailed. */
    KnownScheduledActionKind["Email"] = "Email";
    /** Cost anomaly information will be emailed. Available only on subscription scope at daily frequency. If no anomaly is detected on the resource, an email won't be sent. */
    KnownScheduledActionKind["InsightAlert"] = "InsightAlert";
})(KnownScheduledActionKind || (exports.KnownScheduledActionKind = KnownScheduledActionKind = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (exports.KnownCreatedByType = KnownCreatedByType = {}));
/** Known values of {@link CheckNameAvailabilityReason} that the service accepts. */
var KnownCheckNameAvailabilityReason;
(function (KnownCheckNameAvailabilityReason) {
    /** Invalid */
    KnownCheckNameAvailabilityReason["Invalid"] = "Invalid";
    /** AlreadyExists */
    KnownCheckNameAvailabilityReason["AlreadyExists"] = "AlreadyExists";
})(KnownCheckNameAvailabilityReason || (exports.KnownCheckNameAvailabilityReason = KnownCheckNameAvailabilityReason = {}));
/** Known values of {@link LookBackPeriod} that the service accepts. */
var KnownLookBackPeriod;
(function (KnownLookBackPeriod) {
    /** 7 days used to look back. */
    KnownLookBackPeriod["Last7Days"] = "Last7Days";
    /** 30 days used to look back. */
    KnownLookBackPeriod["Last30Days"] = "Last30Days";
    /** 60 days used to look back. */
    KnownLookBackPeriod["Last60Days"] = "Last60Days";
})(KnownLookBackPeriod || (exports.KnownLookBackPeriod = KnownLookBackPeriod = {}));
/** Known values of {@link Grain} that the service accepts. */
var KnownGrain;
(function (KnownGrain) {
    /** Hourly grain corresponds to value per hour. */
    KnownGrain["Hourly"] = "Hourly";
    /** Hourly grain corresponds to value per day. */
    KnownGrain["Daily"] = "Daily";
    /** Hourly grain corresponds to value per month. */
    KnownGrain["Monthly"] = "Monthly";
})(KnownGrain || (exports.KnownGrain = KnownGrain = {}));
/** Known values of {@link Term} that the service accepts. */
var KnownTerm;
(function (KnownTerm) {
    /** Benefit term is 1 year. */
    KnownTerm["P1Y"] = "P1Y";
    /** Benefit term is 3 years. */
    KnownTerm["P3Y"] = "P3Y";
})(KnownTerm || (exports.KnownTerm = KnownTerm = {}));
/** Known values of {@link Scope} that the service accepts. */
var KnownScope;
(function (KnownScope) {
    /** Single scope recommendation. */
    KnownScope["Single"] = "Single";
    /** Shared scope recommendation. */
    KnownScope["Shared"] = "Shared";
})(KnownScope || (exports.KnownScope = KnownScope = {}));
/** Known values of {@link BenefitKind} that the service accepts. */
var KnownBenefitKind;
(function (KnownBenefitKind) {
    /** Benefit is IncludedQuantity. */
    KnownBenefitKind["IncludedQuantity"] = "IncludedQuantity";
    /** Benefit is Reservation. */
    KnownBenefitKind["Reservation"] = "Reservation";
    /** Benefit is SavingsPlan. */
    KnownBenefitKind["SavingsPlan"] = "SavingsPlan";
})(KnownBenefitKind || (exports.KnownBenefitKind = KnownBenefitKind = {}));
/** Known values of {@link GrainParameter} that the service accepts. */
var KnownGrainParameter;
(function (KnownGrainParameter) {
    /** Hourly grain corresponds to value per hour. */
    KnownGrainParameter["Hourly"] = "Hourly";
    /** Hourly grain corresponds to value per day. */
    KnownGrainParameter["Daily"] = "Daily";
    /** Hourly grain corresponds to value per month. */
    KnownGrainParameter["Monthly"] = "Monthly";
})(KnownGrainParameter || (exports.KnownGrainParameter = KnownGrainParameter = {}));
